### ISSUE TITLE
Firejailed Gwenview cannot open filenames with blanks

### ISSUE THREAD

**Issue by curiosity-seeker** [netblue30/firejail#58]
I've noticed a strange problem with my firejailed VLC. Whenever I try to open, e.g., a video whose filename contains a blank or is located in a directory with a blank, VLC opens and I get an error like the following (for the file "/media/Multimedia/Filme/Die Tribute von Panem - The Hunger Games.avi"):
![panem](https://cloud.githubusercontent.com/assets/14075215/9736722/3c135db4-5643-11e5-8807-fca8394a7999.png)
Note that this problem does **not** occur if I open VLC directly and navigate to that file, and it does **not** occur with an un-firejailed VLC. Rather, it only occurs if I want to open that file with firejailed VLC in a filemanager (Krusader or Dolphin - I'm running KDE Plasma 5 on Arch Linux). A similar problem happens if I want to open pictures with blanks in their filenames in firejailed Gwenview.

**Comment by pmillerchip** [comment]
If you could please post the exact command line that you use to start firejail, we can help diagnose the problem.

It sounds like a situation where firejail is getting started from a shortcut like `firejail vlc $1` - in this case the `$1` gets expanded without protecting the spaces in the filename. Is it something like that?

**Comment by curiosity-seeker** [comment]
Thanks for your reply. The exact command line in the KDE menu entry is: 

firejail --profile=/home/xxx/.config/firejail/vlc.profile /usr/bin/vlc --started-from-file %U

**Comment by curiosity-seeker** [comment]
FWIW, I've just tried to start that video without the

 --started-from-file %U

switches but with the the same result.

**Comment by netblue30** [comment]
Add a --shell=none, like this:

`````
firejail --shell=none --profile=/home/xxx/.config/firejail/vlc.profile /usr/bin/vlc --started-from-file %U
`````

**Comment by curiosity-seeker** [comment]
Wow! That solved it, indeed! Thank you very much! Can you explain why?

However, that similar problem with Gwenview is not yet really solved. The command line for it is:

firejail --profile=/home/xxx/.config/firejail/gwenview.profile gwenview %U

When opening pictures with blank(s) Gwenview opens with just a line that say something like

Basisordner > tmp > org.kde.gwenview-hNwKkl

After adding --shell=none Gwenview displays the folder symbol where the picture is located and I can navigate to it. An improvement, indeed, but the picture itself isn't immediately opened as it should. Do you have an idea how to circumvent this problem?

**Comment by netblue30** [comment]
It all depends on how the file manager starts vlc. If they start the program in a shell, and firejail also using a shell to start the program, between the two shells the file name gets split into several file names. So the fix is not to let firejail to start a second shell.

I'll look into gwneview, thanks for the bug.

**Comment by curiosity-seeker** [comment]
FWIW, --shell=none also solves the same problem for a firejailed LibreOffice. Just as a reference for a possible new issue ... ;-)

**Comment by netblue30** [comment]
Thanks!

**Comment by curiosity-seeker** [comment]
I suggest that you add

shell none

to the deafult VLC profile as it solves this issue. Or are you looking for a general solution as this issue is still open?

**Comment by netblue30** [comment]
Thanks, I'll look for a more general solution, otherwise I'll end up with "shell none" allover the place.

**Comment by curiosity-seeker** [comment]
I found [this](https://www.wilderssecurity.com/threads/firejail-linux-sandbox.369309/page-8#post-2536579) post which seems to be relevant for the problem discussed here. I can confirm that `shell none` does no longer solve the problem with the newest gwenview version but the start script in that post does!

**Comment by netblue30** [comment]
Thanks, I'll take a look.

**Comment by brunonova** [comment]
I also noticed this issue when I double-clicked an HTML file. The sandoxed Firefox failed to open the file. Opening the file from the command-line didn't work.
Adding `shell none` to the profile fixes the issue.

This doesn't work:

    firejail vlc "some file.mp4"

But this works:

    firejail vlc "\"some file.mp4\""

When the program is started through a shell, could you add `"` or `'` characters around every argument?

**Comment by netblue30** [comment]
I think this is fixed in new version, 0.9.36.
